packages/backend-core/src/features/index.ts

```diff
@@ -150,6 +156,26 @@ export class FlagSet<V extends Flag<any>, T extends { [key: string]: V }> {
         }
       }
 
+      const license = ctx?.user?.license
+      if (license) {
+        for (const feature of license.features) {
+          const flag = this.flags[feature]
+          if (!flag) {
+            continue
+          }
+
+          if (flags[feature] === true || specificallySetFalse.has(feature)) {
```

Comment:
NAB: I don't understand the variable `specificallySetFalse` . What is that?

Reply:
When an environment variable specifies a _negative_ feature flag, e.g. `TENANT_FEATURE_FLAGS=tenant1:!DEFAULT_VALUES` they're wanting to disable `DEFAULT_VALUES`. This `specificallySetFalse` set tracks those flags and makes sure that later steps in the flag fetching process don't set them back to true.

It's up for debate whether this is the correct behaviour or not, but I went with it because this negative flag behaviour already exists and might be in use by self-hosters. I think a flag suddenly becoming enabled when they're set it false on purpose could be a bad experience.

Reply:
Makes sense!